src/main/java/com/capgemini/wsb/fitnesstracker/training/internal/TrainingController.java

```diff
@@ -0,0 +1,75 @@
+package com.capgemini.wsb.fitnesstracker.training.internal;
+
+import com.capgemini.wsb.fitnesstracker.training.api.dto.*;
+import com.capgemini.wsb.fitnesstracker.training.api.Training;
+import com.capgemini.wsb.fitnesstracker.user.api.UserService;
+import com.capgemini.wsb.fitnesstracker.user.api.dto.UserDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+@RestController
+@RequestMapping("/v1/trainings")
+@RequiredArgsConstructor
+public class TrainingController {
+    private final UserService userService;
+    private final TrainingServiceImpl trainingService;
+    private final TrainingMapper trainingMapper;
+    private final TrainingRepository trainingRepository;
+
+    @GetMapping
+    public ResponseEntity<List<TrainingDto>> getAllTrainings() {
+        return ResponseEntity.ok(trainingService.findAllTrainings());
+    }
+
+    @GetMapping("/user")
+    public List<Training> getTrainingsByUser(@RequestParam Long userid) {
+        UserDto user = userService.findUserById(userid);
+        return trainingService.getTrainingsByUser(user);
+    }
+
+    @GetMapping("/user/{userId}")
+    public ResponseEntity<List<TrainingDto>> getTrainingsByUserId(@PathVariable Long userId) {
+        UserDto user = userService.findUserById(userId);
+        List<TrainingDto> trainings = trainingService.getTrainingsByUser(user).stream()
```

Comment:
Mapping entity to dto and reverse should be done inside service method, not in the controller.

Reply:
Please map object inside service.